web/src/stream_popover.ts

```diff
@@ -500,6 +504,90 @@ export async function build_move_topic_to_stream_popover(
             is_disabled;
     }
 
+    // Warn if any of current topic participants are NOT subscribed
+    // to the destination stream.
+    function warn_unsubscribed_participants(destination_stream_id: number): void {
+        $("#move_topic_modal .unsubscribed-participants-warning").remove();
+
+        const conversation_messages = message_util.get_messages_in_topic(
+            current_stream_id,
+            topic_name,
+        );
+
+        const active_human_participants = new ConversationParticipants(
+            conversation_messages,
+        ).visible();
+        const unsubscribed_participant_ids = [...active_human_participants].filter(
+            (user_id) => !stream_data.is_user_subscribed(destination_stream_id, user_id),
+        );
+
+        const n_unsubscribed_participants = unsubscribed_participant_ids.length;
+
+        if (n_unsubscribed_participants === 0) {
+            // This is true in the following cases, for all of which we do nothing :
+            // 1- Conversation (topic) has no participants.
+            // 2- All participants are subscribed to the destination stream.
+            // 3- Destination stream is current stream.
+            return;
+        }
+
+        const unsubscribed_participant_names_list = util.format_array_as_list_with_conjuction(
+            unsubscribed_participant_ids.map(
+                (user_id) =>
+                    `<strong>${people.get_user_by_id_assert_valid(user_id).full_name}</strong>`,
```

Comment:
Never insert user-controlled data like names into a JavaScript string template and then pass it as HTML without escaping, it's an XSS bug.

Reply:
Yes, I will use `format_array_as_list_with_highlighted_elements` as it bolds the names and provides escaping.

And to prevent assumtions, I tested it by including `<script>alert("hacked !")</script>` in one of the names, 
with `format_array_as_list_with_conjuction` the script was executed showing the alert, while `format_array_as_list_with_highlighted_elements` just rendered the names.